jmdict.py

```diff
@@ -404,85 +410,89 @@ def parse_translation(self):
-    # n for jmnedict
-    # c for the combined
-    # the default is jmdict
-    elif(arg == '-d'):
-        for c in sys.argv[i+1]:
-            if(c == "j"):
-                create_jmdict = True
-            elif(c == "n"):
-                create_jmnedict = True
-            elif(c == "c"):
-                create_combined = True
-        i += 1
-    # The -p flag enables pronunciations
-    elif(arg == '-p'):
-        pronunciations = True
-        i+=1
-
-    i += 1
-
-if(not create_combined and not create_jmdict and not create_jmnedict):
-    create_jmdict = True
-
-#Create files
-if(create_jmdict or create_combined):
-    sys.stderr.write('Parsing JMdict_e.gz...\n')
-    parser = JMdictParser('JMdict_e.gz')
-    jmdict_entries = parser.parse()
-    if(pronunciations):
-        sys.stderr.write('Adding pronunciations...\n')
-        ac = Pronunciation()
-        count = ac.addPronunciation(jmdict_entries)
-        sys.stderr.write(f"added {count} pronunciations\n")
-    sys.stderr.write('Created %d entries\n' %len(jmdict_entries))
-        
-    if(max_sentences > 0):
-        sys.stderr.write('Adding sentences...\n')
-        examples = ExampleSentences("jpn_indices.tar.bz2", "sentences.tar.bz2", jmdict_entries)
-        sys.stderr.write("Sentences added: " + str(examples.addExamples(only_good_sentences, max_sentences)) + "\n")
-
-if(create_jmdict):
-    sys.stderr.write('Creating files for JMdict...\n')
-    write_index(jmdict_entries, "jmdict", "JMdict Japanese-English Dictionary", sys.stdout, default_index=VOCAB_INDEX)
-
-if(create_jmnedict or create_combined):
-    sys.stderr.write('Parsing JMnedict.xml.gz...\n')
-    parser = JMnedictParser('JMnedict.xml.gz')
-    jmnedict_entries = parser.parse()
-    sys.stderr.write('Created %d entries\n' %len(jmnedict_entries))
-
-if(create_jmnedict):
-    sys.stderr.write('Creating files for JMnedict...\n')
-    write_index(jmnedict_entries, "jmnedict", "JMnedict Japanese Names", sys.stdout, default_index=NAME_INDEX)
-
-if(create_combined):
-    sys.stderr.write('Creating files for combined dictionary\n')
-    write_index(jmdict_entries+jmnedict_entries, "combined", "JMdict Japanese-English Dictionary and JMnedict Japanese Names", sys.stdout, default_index=None)
+def get_args():
+
+    class DictAction(argparse.Action):
+        def __init__(self, option_strings, dest, nargs=None, **kwargs):
+             if nargs is not None:
+                raise ValueError("nargs not allowed")
+             super().__init__(option_strings, dest, **kwargs)
+        def __call__(self, parser, namespace, values, option_string=None):
+            setattr(namespace, self.dest, SimpleNamespace(create_jmdict = 'j' in values, create_jmnedict = 'n' in values, create_combined = 'c' in values))
+
+    arg_parser = argparse.ArgumentParser(description='Build the JMdict or JMnedict for Kindle', formatter_class=argparse.RawTextHelpFormatter)
+    arg_parser.add_argument('-s', '--sentences'
+        ,action='store'
```

Comment:
I don't want to nitpick, but commas like this look awkward to me.

I'm starting to think it might be better to start using some sort of automated Python code formatting tool to ensure code is consistent.

Reply:
Trailing commas.  https://www.python.org/dev/peps/pep-0008/#when-to-use-trailing-commas

Honestly, I think the easiest thing to do is to use an automated tool to format like https://www.python.org/dev/peps/pep-0008/  .  I know there are a few, and I even recall to use one a couple of years ago, but I forgot which.